### PR TITLE
KAFKA-9921: explicit handling of null values with retainDuplicates

### PR DESCRIPTION
**Pull request by ableegoldman**
In general the behavior of window stores with `retainDuplicates` is not well documented or enforced, so we should attempt to clarify things better in the javadocs and in the code itself. This explicitly skips the put/delete when the value is null and duplicates are allowed, and specifies this behavior in the docs.

Also adds in a test I left out in the earlier PR https://github.com/apache/kafka/pull/8564

### CODE REVIEW

**Comment by mjsax** [comment]
Retest this please

File: streams/src/main/java/org/apache/kafka/streams/state/WindowStore.java [hunk 1]
```diff
@@ -57,6 +57,10 @@
 
     /**
      * Put a key-value pair into the window with given window start timestamp
+     * <p>
+     * If serialized value bytes are null it is treated as delete. Note that as an optimization, deletes will be
+     * skipped in the case of an underlying store that allows duplicates.
```
Comment (GeorgiPetkov) [inline streams/src/main/java/org/apache/kafka/streams/state/WindowStore.java]: I would use the 'interpreted' instead of 'treated' to be more consistent with the rest of the javadoc (at least in this file). The behavior is not really due to an optimization and that's not really the point IMO. Also, 'retain duplicates' in the more common term instead of 'allow duplicates'. I would rephrase it to something like this:
```suggestion
     * If serialized value bytes are null it is treated as delete. Note that deletes will be
     * ignored in the case of an underlying store that retains duplicates.
```

Comment (GeorgiPetkov) [reply streams/src/main/java/org/apache/kafka/streams/state/WindowStore.java]: lol, I forgot to apply my own suggestion to use 'interpreted', good catch.

**Comment by mjsax** [comment]
Retest this please

**Review by mjsax — APPROVED** [review]
LGTM (assuming Jenkins passes).

**Comment by mjsax** [comment]
Retest this please.

**Comment by ableegoldman** [comment]
Java 14 passed, Java 8 and Java 11 has one failure each but results cleaned up already.

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `2.5`.
